core/src/main/java/com/spotify/ffwd/output/CoreOutputManager.java

```diff
@@ -181,6 +200,34 @@ public final Long getHLLPRefreshPeriodLimit() {
         } else {
             this.cardinalityLimit = null;
         }
+
+        if (!Strings.isNullOrEmpty(dynamicTagsFile)) {
+            final File file = new File(dynamicTagsFile);
+            if (file.exists() && file.canRead()) {
+                final int initialDelay =
+                    System.getenv().containsKey("IS_FFWD_CONFIGURATION_TEST") ? 3 : 60;
+                tagRefreshingThread = Executors.newSingleThreadScheduledExecutor(
+                    new ThreadFactoryBuilder().setNameFormat("tag-refreshing-thread").build());
+                tagRefreshingThread.scheduleWithFixedDelay(() -> refreshTagsFromFile(file.toPath()),
+                    initialDelay, 60, TimeUnit.SECONDS);
+            } else {
+                log.warn("Cannot read file at {}. Ignoring it.", dynamicTagsFile);
+            }
+        }
+    }
+
+    private void refreshTagsFromFile(final Path path) {
+        try (final Stream<String> lines = Files.lines(path)) {
+            lines.forEach(label -> {
+                final String[] labelParts = label.split("=", 2);
```

Comment:
Assumption here is that tag key and values are separated by `=`.

Reply:
I'm curious if instead of this approach we should try to use yaml style configuration as we have in standard Apollo?
https://github.com/lightbend/config
Based on examples the labels.yaml file is yaml and I'm curious if we going to have more control of the labels.

Reply:
Which labels.yaml are you referring to?

The K8s downward API doesn't let you configure the format of the file it creates [1]. It'll always be like this.

```
key1=val1
key2=val2
```

1: https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/#store-pod-fields

Reply:
sounds good.